src/scripting/game_lua_kernel.cpp

```diff
@@ -2581,6 +2580,7 @@ int game_lua_kernel::intf_put_unit(lua_State *L)
 		}
 
 		unit_ptr u = unit::create(cfg, true, vcfg);
+		units().erase(loc);
```

Comment:
Is this code path deprecated? It's too late in the 1.18 cycle to remove it, but putting the location before the unit seems to be undocumented in the LuaAPI Wiki pages.

I tried adding an assert, and it didn't trigger when I ran all the unit tests.

Reply:
Putting the location before the unit was the old way of doing it. It was flipped in order to support the colon calling convention.

So yes, that's deprecated.

I'm not sure how this particular line of code relates to that deprecated call mode though.

Reply:
Thanks for the explanation about why it was flipped.

> I'm not sure how this particular line of code relates to that deprecated call mode though.

If I'm reading the code correctly, this line is never reached when the first argument is a unit, so it's only for the deprecated call. However, the line's just moved in this PR, it isn't a change to what the deprecated path does.

Reply:
> If I'm reading the code correctly, this line is never reached when the first argument is a unit

It stil lcreates a unit from the first argument, so its the path that constructs a unit (as first argeument ) and places it on the map,

Reply:
First argument being a location is deprecated. First argument being a config describing a unit is not; it is, in fact, precisely the new format for creating a _new_ unit. (Old format was location, config.)